conans/client/cmd/search.py

```diff
@@ -11,23 +11,36 @@ def __init__(self, client_cache, remote_manager, remote_registry):
 
     def search_recipes(self, pattern, remote=None, case_sensitive=False):
         ignorecase = not case_sensitive
+
+        references = {}
         if not remote:
-            return search_recipes(self._client_cache, pattern, ignorecase)
+            references[None] = []
+            refs = search_recipes(self._client_cache, pattern, ignorecase)
+            for ref in refs:
+                references[None].append((ref, self._get_recipe_hash_local(ref)))
+            return references
 
         if remote == 'all':
             remotes = self._registry.remotes
             # We have to check if there is a remote called "all"
             # Deprecate: 2.0 can remove this check
             if 'all' not in (r.name for r in remotes):
-                references = {}
                 for remote in remotes:
-                    result = self._remote_manager.search_recipes(remote, pattern, ignorecase)
-                    if result:
-                        references[remote.name] = result
-                return references
+                    refs = self._remote_manager.search_recipes(remote, pattern, ignorecase)
+                    if refs:
+                        references[remote.name] = []
+                        for ref in refs:
+                            recipe_hash = self._get_recipe_hash_remote(ref, remote)
+                            references[remote.name].append((ref, recipe_hash))
+                return OrderedDict(references)
         # single remote
         remote = self._registry.remote(remote)
-        return self._remote_manager.search_recipes(remote, pattern, ignorecase)
+        references[remote.name] = []
+        refs = self._remote_manager.search_recipes(remote, pattern, ignorecase)
+        for ref in refs:
+            recipe_hash = self._get_recipe_hash_remote(ref, remote)
+            references[remote.name].append((ref, recipe_hash))
+        return OrderedDict(references)
```

Comment:
It doesn't make sense, sometimes you return an OrderedDict, other you return a dict.

Building an OrderedDict from a dict makes no sense (regarding ordering)